docs/tools/firmware.sh

```diff
@@ -18,36 +18,178 @@ while getopts "vhx" option; do
 	esac
 done
 
+prerequisite_packages () {
+cat <<EOF
+
+Prerequisites needed to continue:
+
+1. Python 3
+2. ${additional_requirement}
+
+Python 3 is a part of Xcode command line tools. You can also use Homebrew, MacPorts etc.
+Continue when you have the prerequisites installed.
+
+Continue? (Y/n)
+EOF
+}
```

Comment:
Only ask people to install stuff if they dont already have it:

```
if ! command -v COMMAND &> /dev/null; then
    # ask to install it here, then return a error
fi
```

Also do this where you ask people to install python later in the script. You don't need to ask people to continue, just exit if the command is not available.